Support of golang types Time, Uint and Uint64 added

hus787 opened the pull request:
Minor changes: iota used instead of defining const values by hand

hus787 committed:
iota used instead of defining const values by hand

hus787 committed:
support Time, Uint and Uint64 added

dennwc commented on type.go:
```diff
@@ -12,35 +12,40 @@ type OType byte
-	LINKBAG      OType = 22
-	ANY          OType = 23
-	UNKNOWN      OType = 255 // driver addition
+	BOOLEAN OType = iota
```
As a general rule, you should not use iota to reflect constants defined in third-party library.

hus787 replied:
Agreeed. Shall revert.

Wish we could do something like: https://github.com/lib/pq/blob/1046c2c577f5be9d829bda808a5696c4c74c9f3b/oid/gen.go#L51

dennwc commented on type.go:
```diff
@@ -12,35 +12,40 @@ type OType byte
-const intSize = unsafe.Sizeof(int(0))
+const (
+	intSize  = unsafe.Sizeof(int(0))
+	uintSize = unsafe.Sizeof(uint(1))
```
Why to use both? Is it possible to hit the case when `intSize != uintSize`?

hus787 replied:
Since the size is implementation detail, then it's better to play safe.

dennwc replied:
I'd better to add test for this extra-rare case if you like :) It's architecture-dependent, and I'm, not aware of any architectures with different sizes for uint and int.

hus787 committed:
reverting the use of iota

https://github.com/istreamdata/orientgo/pull/55/files/5d7b1ae5d826f9ca61d3b48cc31a0622067c466d#r60765765

dennwc committed:
Bump Go version

dennwc commented:
I have fixed CI, it should work now.
When all comments will be addressed, please squash your commits and rebase it on top of current v2 branch.

dennwc commented:
Merged as a1fc07b5a7e42c45d849702d0f3c50c2e0f60a1d
